### PR TITLE
Feature/admin item crud

### PR DESCRIPTION
**Pull request by Jguerra47**
- Added all the model Item and its functions
- Added all the migration and seeders files to replicate its table
- Design views related to Item CRUD
- Use session vars to manage creation
- Lang support to EN
- ImageStorage to handle image input

_1. List items view:_
image

_2. Create items view:_
image

_3. Show items view:_
image

Notes: Could be added spans to clarify the kind of attributes related to an Item but it can be checked in a UX release.

### CODE REVIEW

File: lang/en/actions.php [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?php
+
+return [
+    'send' => 'Send',
+    'delete' => 'Delete',
+];
```
Comment (Juanda0) [inline lang/en/actions.php:6]: Add last whitespace

File: lang/en/commons.php [hunk 2]
```diff
@@ -1,18 +1,34 @@
...
-    'devs' => 'The KB devs',
     'comment' => 'comment',
     'rating' => 'rating',
     'ratingSpan' => 'Rating: :value/5',
     'commentSpan' => 'comment: :value',
-    'review' => 'Review #:id',
-    'reviewTitle' => 'Reviews section',
-    'reviewList' => 'List of reviews'
+    'review' => 'Review #:id'
 ];
```
Comment (Juanda0) [inline lang/en/commons.php:34]: idem

File: resources/views/admin/item/create.blade.php [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+                                    value="{{ old('stock') }}" />
+                                <input type="submit" class="btn btn-primary" value={{__('actions.send')}} />
+                            </form>
+                        </div>
+                    </div>
+                </div>
+            </div>
+        </div>
+    </div>
+@endsection
```
Comment (Juanda0) [inline resources/views/admin/item/create.blade.php:49]: idem

File: resources/views/admin/item/index.blade.php [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+                                <path d="M16 8s-3-5.5-8-5.5S0 8 0 8s3 5.5 8 5.5S16 8 16 8zM1.173 8a13.133 13.133 0 0 1 1.66-2.043C4.12 4.668 5.88 3.5 8 3.5c2.12 0 3.879 1.168 5.168 2.457A13.133 13.133 0 0 1 14.828 8c-.058.087-.122.183-.195.288-.335.48-.83 1.12-1.465 1.755C11.879 11.332 10.119 12.5 8 12.5c-2.12 0-3.879-1.168-5.168-2.457A13.134 13.134 0 0 1 1.172 8z"/>
+                                <path d="M8 5.5a2.5 2.5 0 1 0 0 5 2.5 2.5 0 0 0 0-5zM4.5 8a3.5 3.5 0 1 1 7 0 3.5 3.5 0 0 1-7 0z"/>
+                            </svg>
+                        </a>
+                    </div>
+                </div>
+            </div>
+        @endforeach
+    </div>
+@endsection
```
Comment (Juanda0) [inline resources/views/admin/item/index.blade.php:24]: idem

File: resources/views/admin/item/show.blade.php [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+                </div>
+            </div>
+            <form method="POST" action="{{ route('admin.item.delete', ['id'=> $viewData['item']->getId()]) }}">
+                @csrf
+                @method("DELETE")
+                <input type="submit" class="btn btn-danger" value={{__('actions.delete')}}>
+            </form>
+        </div>
+    </div>
+@endsection
```
Comment (Juanda0) [inline resources/views/admin/item/show.blade.php:35]: idem

**Review by Juanda0 — APPROVED** [review]

File: database/migrations/2023_03_20_011540_create_items_table.php [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::create('items', function (Blueprint $table) {
+            $table->id();
+            $table->string('title');
+            $table->string('type');
+            $table->integer('price');
```
[the remark this answers is not in the document]
Comment (Jguerra47) [reply database/migrations/2023_03_20_011540_create_items_table.php:18]: Taking into consideration the professor's advice on price data management, the class diagram should be updated to take an integer value avoiding precision issues.

An useful video about that topic: https://www.youtube.com/watch?v=UssS8GIU9TI

File: app/Http/Controllers/ItemController.php [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+        $viewData['items'] = Item::all();
+
+        return view('admin.item.index')->with('viewData', $viewData);
+    }
+
+    public function show(string $id): View|RedirectResponse
+    {
+        $viewData = [];
+        $item = Item::find($id);
+        if (is_null($item)) {
```
Comment (Juanda0) [inline app/Http/Controllers/ItemController.php:25]: Try to implement the try-catch alternative
```suggestion
       try {
            $viewData = [];
            $item = Item::findOrFail($id);
            $viewData['item'] = $item;

            return view('admin.item.show')->with('viewData', $viewData);
        } catch (Exception $e) {
            return redirect()->route('admin.index');
        }
```

File: app/Providers/ImageServiceProvider.php [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+
+namespace App\Providers;
+
+use App\Interfaces\ImageStorage;
+use App\Util\ImageLocalStorage;
+use Illuminate\Support\ServiceProvider;
+
+class ImageServiceProvider extends ServiceProvider
+{
+    public function boot()
```
Comment (DanielHernandezO) [inline app/Providers/ImageServiceProvider.php:11]: Delete the "boot" function, because it has nothing

**Review by Juanda0 — APPROVED** [review]

**Pull request merged**